src/TensorKit.jl

```diff
@@ -183,6 +184,7 @@ include("spaces/vectorspaces.jl")
 include("tensors/abstracttensor.jl")
 # include("tensors/tensortreeiterator.jl")
 include("tensors/tensor.jl")
+include("tensors/diagtensor.jl")
```

Comment:
As a suggestion, I feel like `diagonal.jl` is more in line with `adjoint.jl` for the filenames.